### DISCUSSION TITLE
Change default swagger example value

### DISCUSSION THREAD

**Discussion by vinicius-oa** [fastapi/fastapi#10122] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/", status_code=202)
async def _():
    raise NotImplementedError
```

### Description

Is it possible to change the example value to None? Or hide it? My resource won't return a response body.
In OpenAPI docs it says you can define a empty response, but I tried its format by passing it as a dict to `app.post` decorator's responses parameter, but it didn't work.

![image](https://github.com/tiangolo/fastapi/assets/139140185/cd477059-a443-4c3a-afe7-9e5fa59ecd15)

### Operating System

Linux

### FastAPI Version

0.101.1

### Pydantic Version

2.2.1

### Python Version

3.11.4

**Comment by ebottos94** [comment]
Hi @vinicius-oa, you can do something like this : 

```python
from fastapi import FastAPI
from pydantic import BaseModel

class NullResponseModel(BaseModel):
  class Config:
    schema_extra = {
      "example" : {None}
    }

app = FastAPI()


@app.post("/", responses={202 : {"model": NullResponseModel}}, status_code=202)
async def null_response():
  return None
```

**Reply by Kludex** [reply, marked as answer]
With Pydantic V2, you can do as follows:

```py
from typing import Annotated
from fastapi import FastAPI
from pydantic import WithJsonSchema


app = FastAPI()

Null = Annotated[None, WithJsonSchema(json_schema={"type": "null"})]


@app.post("/", response_model=Null, status_code=202)
async def null_response() -> None:
    return None
```

But... I think we need to investigate this. FastAPI itself may be the one to infer this.

**Reply by mikenerone** [reply]
Unfortunately, this isn't sufficient if you want to return a `204 NO CONTENT` (with FastAPI 0.110.0). Setting `response_model` to anything but `None` in combination with `status_code=204` results in a crash error:
```
  File ".../.venv/lib/python3.12/site-packages/fastapi/routing.py", line 468, in __init__
    assert is_body_allowed_for_status_code(
AssertionError: Status code 204 must not have a response body
```
A `204` is an obvious case were having an example value at all is not appropriate.

**Reply by vincentfenet** [reply]
Still the same exact problem with this code:

```
@router.put("/test", status_code=status.HTTP_204_NO_CONTENT)
async def test():
    pass
```

What should we do ?
